src/ml_tooling/baseclass.py

```diff
@@ -139,20 +163,47 @@ def save_estimator(self, path: str) -> pathlib.Path:
                     "You haven't scored the estimator - no results available to log"
                 )
 
-            metric_scores = {self.result.metric: float(self.result.score)}
-
-            log_results(
-                metric_scores=metric_scores,
-                estimator_name=self.estimator_name,
-                estimator_params=self.result.estimator_params,
-                run_dir=self.config.RUN_DIR,
-                estimator_path=str(estimator_file),
-            )
+            log = Log.from_result(self.result, estimator_path=estimator_file)
+            log.save_log(self.config.RUN_DIR)
 
         logger.info(f"Saved estimator to {estimator_file}")
 
         return estimator_file
 
+    def dump(self):
+        if self.is_pipeline:
+            return [
+                {
```

Comment:
How can you tell whici parts of the pipe connects where?

Reply:
`pipeline.steps` is a list of transformations, so the order is fixed

Reply:
We should test that this works with multiple pipliens and unions.

Reply:
Added testcases for Unions and changed code to recursively look in params for Pipelines